### ISSUE TITLE
Python runs into "exception: access violation reading 0xFFFFFFFFFFFFFFFF"

### ISSUE THREAD

**Issue by ffreemt** [alifcommunity/webui#47]
```bash
2023-01-28 13:09:25.610 | ERROR    | __main__:slot_tab1:226 - exception: access violation reading 0xFFFFFFFFFFFFFFFF
Traceback (most recent call last):

  File "C:\deepl-tr-webui\.venv\lib\site-packages\webui\webui.py", line 88, in events
    self.cb_fun_list[element_id](e)
    │    │           │           └ <webui.webui.event object at 0x000002B1C393D790>
    │    │           └ 2
    │    └ [64, <function close_the_application at 0x000002B1C3868820>, <function slot_tab1 at 0x000002B1C3868940>, <function slot_tab2 ...
    └ <webui.webui.window object at 0x000002B1C38EB700>

> File "C:\mat-dir\pypi-projects\deepl-tr-webui\deepl_tr\__main__.py", line 222, in slot_tab1
    evt.window.show(_)
    │   │      │    └ '\n\n<!DOCTYPE html>\n<html>\n    <head>\n        <title>deepl-tr-webui v0.1.0a1</title>\n\n\t\t<meta charSet="UTF-8"/>\n\t\t...
    │   │      └ <function window.show at 0x000002B1C3240790>
    │   └ <webui.webui.window object at 0x000002B1C38EB700>
    └ <webui.webui.event object at 0x000002B1C393D790>

  File "C:\deepl-tr-webui\.venv\lib\site-packages\webui\webui.py", line 109, in show
    WebUI.webui_show_cpy(self.window, html.encode('utf-8'), 0)
    │     │              │    │       │    └ <method 'encode' of 'str' objects>
    │     │              │    │       └ '\n\n<!DOCTYPE html>\n<html>\n    <head>\n        <title>deepl-tr-webui v0.1.0a1</title>\n\n\t\t<meta charSet="UTF-8"/>\n\t\t...
    │     │              │    └ c_void_p(2962450691904)
    │     │              └ <webui.webui.window object at 0x000002B1C38EB700>
    │     └ <_FuncPtr object at 0x000002B1C39225F0>
    └ <CDLL 'C:\deepl-tr-webui\.venv\lib\site-packages\webui\webui-2-x64.dll', handle 7ffe91cd0000 at 0x2b1c3...

OSError: exception: access violation reading 0xFFFFFFFFFFFFFFFF
[nodemon] app crashed - waiting for file changes before starting...
```
I use the newest python binding of webui2. Very often when I try to swap html (my ui implments some tabs, each tab is a some html), webui crashes with the error message above. The ``webui-2-x64.dll', handle 7ffe91cd0000 at 0x2b1c3...`` part seems to be the same in each crash. My environ: Win10, python 3.8.10, webui2 2.0.3 (pypi).

Is there anything I can do to fix it?  Thanks a lot.

**Comment by hassandraga** [comment]
Thank you for sharing this. I'm not sure, but I guess there is a mix between the WebUI **v2.0.3** in the [PyPI package](https://pypi.org/project/webui2/) and WebUI **v2.0.5** in this [GitHub repo](https://github.com/alifcommunity/webui). 

 `C:\deepl-tr-webui\...\webui-2-x64.dll` should be the same version as `C:\deepl-tr-webui\...\webui.py`

For now, try to building WebUI source code using any C compiler to get the latest `webui-2-x64.dll`, I will update the PyPI package in the next few days after finishing implementing some new features.

**Comment by ffreemt** [comment]
Thanks very much for the reply. I look forward to pypi webui2 v2.0.5.

I seem to have successfully compiled build/Windows/MSVC. But when I try to run ``main.py`` in ``webui\examples\Python\dev``, I got
```
$ python main.py
WebUI Exception: [WinError 193] %1 is not a valid Win32 application
```
I also copied  ` webui-2-static-x64.lib  webui-2-x64.dll  webui-2-x64.lib` from `build/Windows/MSVC` and the webui.py in the repo to `.venv\Lib\site-packages\webui` and of course still got the message.

I guess I'll just wait for the official pypi webui2 v2.0.5.

**Comment by mohamedke** [comment]
I think your system is 32-bit
and the library webui-2-x64.dll is 64-bit
check it out.
If that, You must compile the library in 32-bit
Sorry for my bad english

**Comment by ffreemt** [comment]
Thanks for the hint. But my system is actually a 64bit:
```bash
wmic os get osarchitecture
OSArchitecture
64-bit
```

**Comment by AlbertShown** [comment]
I tried to reproduce the error using https://github.com/ffreemt/deepl-tr-webui but it always works. Probably you have a non-committed code in your machine that has this issue.

**Comment by ffreemt** [comment]
Hm... I tried again, still a lot of crashes with similar exceptions. I was not able to fix it with what I thought might cause the problem and had to put the little side project on hold. 

I've pushed everything to the github repo. If you can find the time, please give it another try. Maybe try to click the Translate tab, edit a cell/save it and click another tab. It will most likely crash.

Maybe I should try a different Python version. I use Python 3.8.10. What python version do you have? Thanks.

**Comment by AlbertShown** [comment]
Sure, I will give it a try soon. I'm using Python `3.10.5`

**Comment by AlbertShown** [comment]
Hi @ffreemt, I tried many times to reproduce the issue but I can't. Those are the steps I did:

1. Run Windows Sandbox
2. Install Python 3.8.10 (default settings) - https://www.python.org/ftp/python/3.8.10/python-3.8.10-amd64.exe
3. Install Git (default settings) - https://git-scm.com/download/win
4. Install GCC TDM  (default settings) - https://jmeubank.github.io/tdm-gcc/download/
5. Install VS Code
6. Terminal -> `pip install git+https://github.com/ffreemt/deepl-tr-webui`
7. Terminal -> `cd Downloads\webui-main\webui-main\build\Windows\GCC` -> `mingw32-make debug`
8. Copy `webui-2-x64.dll` to `AppData\Local\Programs\Python\Python38\Lib\site-packages\webui`
9. Copy `webui-main\packages\PyPI\src\webui\webui.py` to `AppData\Local\Programs\Python\Python38\Lib\site-packages\webui`
10. Terminal -> `git clone https://github.com/ffreemt/deepl-tr-webui`
11. Terminal -> `cd deepl-tr-webui`
12. Terminal -> `python -m deepl_tr`

**Comment by AlbertShown** [comment]
![debug1](https://user-images.githubusercontent.com/116365846/218876029-50acc2da-3853-426c-9172-46e970927f36.png)

**Comment by AlbertShown** [comment]
![image](https://user-images.githubusercontent.com/116365846/218876251-b5a1ad7b-53b5-4c23-9845-07bb161c6f4d.png)

I noticed that the tabs have an empty ID. Probably the tabs need to have a bigger Z level to bring it to the top.

**Comment by AlbertShown** [comment]
![image](https://user-images.githubusercontent.com/116365846/218877002-a5689631-77f6-45c0-aee3-5e4823aad7e9.png)

I tested manually sending the click event using the console and it works.

**Comment by AlbertShown** [comment]
I guess it will be helpful to identify the issue if you re-build the WebUI in debug mode so you can see the debug info in the console, then try to crash it and share the whole console here.

**Comment by AlbertShown** [comment]
[pre-compiled-webui-debug.zip](https://github.com/alifcommunity/webui/files/10737423/pre-compiled-webui-debug.zip)

You can simply use my built files to save time, just copy the files into the Python package folder. Or re-build it using those steps

1. Install GCC TDM (default settings) - https://jmeubank.github.io/tdm-gcc/download/
2. (_Optional_) To also see the logs in DevConsole in the browser change `_webui_log` to true https://github.com/alifcommunity/webui/blob/e56b1e5ea32313c48e775d1f414c7e88e7c60b96/src/webui.c#L27 
3. Terminal -> `cd Downloads\webui-main\webui-main\build\Windows\GCC` -> `mingw32-make debug`

**Comment by ffreemt** [comment]
Thanks a lot. I give another try.

I tried previously with python 3.10, still crashes too often. I used the official pypi webui2 2.0.3

My Win10 has VSC. I successfully compiled webui2 2.0.5 but was not able to use it (some errors). 

I really wonder what's the reasons for those crashes.

**Comment by AlbertShown** [comment]
> I guess it will be helpful to identify the issue if you **re-build the WebUI in debug mode** so you can see the debug info in the console, then try to crash it and **share the whole console here.**

To identify the reason let's start by capturing the WebUI logs and shared here

**Comment by AlbertShown** [comment]
To enable the logs, just open the Visual Studio example project, and uncomment` #define WEBUI_LOG` in [webui.c](https://github.com/alifcommunity/webui/blob/e56b1e5ea32313c48e775d1f414c7e88e7c60b96/src/webui.c#L18)

**Comment by ffreemt** [comment]
I tried your [pre-compiled-webui-debug.zip](https://github.com/alifcommunity/webui/files/10737423/pre-compiled-webui-debug.zip). I can start `python -m deepl_tr` alright, but i can't switch to 2nd tab. So it appears not working or maybe I missed something.

I noticed there is a `webui-2-x64.so` file in `Lib\site-packages\webui` while your pre-compiled-webui-debug.zip just has `webui.py` and `webui-2-x64.dll`. Also, aftering copying your two files to `Lib\site-packages\webui`, `pip show webui2` still shows `Version: 2.0.3`.

I'll try to follow your 12 steps and give another try. I am not too sure about Step 6 

6. Terminal -> pip install git+https://github.com/ffreemt/deepl-tr-webui

Shouldn't I instead `git clone` webui2 repo to the current directory?

Thanks again.

**Comment by AlbertShown** [comment]
`webui-2-x64.so` is a WebUI library for Linux.
`pip show webui2` will still show the version in the config file which is `2.0.3` even if you update the `.dll` file

The purpose of doing this is so you can see the logs in your terminal then copy all the logs and share it here

**Comment by AlbertShown** [comment]
You can also keep the same old code you have, just uncomment `#define WEBUI_LOG` and re-build, then share here the logs from the terminal.

**Comment by AlbertShown** [comment]
I hope you find some free time to share the console logs to help us fix this issue.

**Comment by ffreemt** [comment]
> I tried your [pre-compiled-webui-debug.zip](https://github.com/alifcommunity/webui/files/10737423/pre-compiled-webui-debug.zip). I can start python -m deepl_tr alright, but i can't switch to 2nd tab.

As I mentioned previously, if I use the files in your pre-compiled-webui-debug.zip, I can‘t switch to other tabs when running `python -m deepl_tr`, hence I cannot reproduce a crash.

I also tried to compile following your steps. When I used the compiled files, I got the same outcome: unable to switch to other tabs (hence unable to crash).

I tried various combinations: comment/uncomment `#define WEBUI_LOG`, `mingw32-make`/`mingw32-make debug`. The result is always the same: unable to switch to other tabs.

Is there anything else I should try, or just wait for the new version of webui2 and hope it will fix my problem?
